pinot-common/src/main/java/com/linkedin/pinot/common/data/DateTimeFormatSpec.java

```diff
@@ -39,11 +44,18 @@
       "granularity must be of format size:timeunit";
   public static final String GRANULARITY_PATTERN_ERROR_STR =
       "granularity must be of format [0-9]+:<TimeUnit>";
+
   public static final String NUMBER_REGEX = "[1-9][0-9]*";
+  /** eg: yyyyMMdd tz(CST) or yyyyMMdd HH tz(GMT+0700) or yyyyMMddHH tz(America/Chicago) **/
+  private static final Pattern SDF_PATTERN_WITH_TIMEZONE = Pattern.compile("^(.+)( tz\\((.+)\\))");
```

Comment:
this is sensitive to spaces right? yymmdd  tz ('UTC') will probably not be able to parse the tz.

Reply:
tz[ ]+\\([.*]+\\)[ ]+$ )  might handle spaces gracefully. please validate the regex.

Reply:
Added space handling in regex. Also trimming timezone and sdf strings.